rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1179,175 +1170,191 @@ public NodeState getNodeState(String nodeUrl) {
         return node.getState();
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public BooleanWrapper defineNodeSource(String nodeSourceName, String infrastructureType, Object[] infraParams,
             String policyType, Object[] policyParams, boolean nodesRecoverable) {
-        if (nodeSourceName == null) {
-            throw new IllegalArgumentException("Node Source name cannot be null");
-        }
-
-        List<Serializable> serializableInfraParams = getSerializableParamsOrFail(nodeSourceName, infraParams);
-        List<Serializable> serializablePolicyParams = getSerializableParamsOrFail(nodeSourceName, policyParams);
 
-        logger.info("Define a node source : " + nodeSourceName);
+        this.validateNodeSourceNameOrFail(nodeSourceName);
 
         nodeSourceName = nodeSourceName.trim();
 
+        logger.info("Define node source " + nodeSourceName + " requested by " + this.caller.getName());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " requested by " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)